### ISSUE TITLE
Swap from Netlify to another hosting platform

### ISSUE THREAD

**Issue by sgibson91** [the-turing-way/the-turing-way#2481]
### Summary

As the Person Who Gets Asked All the Technical Questions™️, I find working with Netlify to be too confusing as it's too much of a black box. Historically, we have used this service because they provided a deploy preview link on Pull Requests; however, this service for us has been broken for some time https://github.com/alan-turing-institute/the-turing-way/issues/2308 (and their support's response has been "it's your problem, not ours" 🙄 ).

Now though, the ReadTheDocs team is developing a GitHub Action that will post deploy preview links to Pull Requests https://github.com/readthedocs/readthedocs-preview To my knowledge, this will be the first alternative docs hosting site with this feature allowing the discussion of "Is Netlify the best tool for hosting The Turing Way?" to be truly viable. @choldgraf also championed the RTD team as being "transparent and collaborative about their strategy, operations, alignment with values, etc". RTD is used consistently throughout the Jupyter/JupyterHub and 2i2c ecosystems as a result.
The inner workings of ReadTheDocs are more clear to me as well since it is "just" a Sphinx build under the hood (whereas I'm still not certain what Netlify is doing).

### What needs to be done?

- [ ] Decide to switch or not
- [ ] Identify technical steps to make the switch
- [ ] Make the switch
- [ ] Redirect `the-turing-way.netlify.app` to `the-turing-way.readthedocs.io`
   - Related discussion: https://github.com/alan-turing-institute/the-turing-way/issues/1962

### Who can help?

* Anyone with knowledge of ReadTheDocs
* cc @choldgraf who made the suggestion

---

#### Things Netlify gives us

Here we have a list of features Netlify gives us. So, if we were to move, we would either have to replace these or live without them,

- Automatic TLS cert generation
- [HTTP redirects](https://book.the-turing-way.org/community-handbook/infrastructure/infrastructure-redirects)
- Build previews from PRs

**Comment by malvikasharan** [comment]
Thanks, Sarah! This is a really interesting problem - as in, we will be required to move from tool to tool as most viable and useful for the project and the community members involved in the maintenance.

There are a few more problems to solve here than you listed:
- "As the Person Who Gets Asked All the Technical Questions" -- I think, one person mustn't be holding this responsibility. Thank you for your patience and careful response to all the questions that are passed your way. With The Turing Way governance and working group model, we hope to build a working group of people who are sharing the responsibility of addressing technical questions. Some of this work can be distributed among the staff members who can triage issues and pass them to the right people such as you, reducing the burden of addressing All the Technical Questions. Tagging @aleesteele here to add this to our agenda for next week's maintenance work.

- "Netlify to be too confusing as it's too much of a black box" - I agree with that. The interface internally can be managed easily but the technical tweaking that is not accessible by their user interface, such as why we don't get directed to the right link upon deployment is hard to address.

- workings of ReadTheDocs are more clear as well since it is a Sphinx build under the hood -- that's a valid point.

- Domain (the-turing-way.readthedocs.io): I think we need to take that leap if/when we switch. However, if we switch from netlify with RTD, assuming that this switch in the future will be unavoidable, we should also redirect platform specific domain to a single the-turing-way.io domain (like it has been brought up by several people in the past).

- Process for deciding to switch? - there are a couple of consultations we will need to do, 1) ask more widely in the TTW community if more people have experience working with RTD vs Netlify, in which case, maybe there is more interest from a larger group to take on technical questions related maintenance roles. 2) Get approval from Kirstie and other members of the core team. This can go as an agenda for the next core team meeting next month if we are not in hurry, or if we need to make a faster decision, we might need to make a decision based on a poll/form gathering thoughts from the core team members.

**Comment by sgibson91** [comment]
Yeah, I am unclear on how this switch would affect the translation team and the personas work for navigating the book so we should definitely acquire more intel on how closely coupled those pieces of work are to Netlify (as opposed to just Jupyter Book) and whether RTD's feature set also supports that work.

**Comment by choldgraf** [comment]
A few quick thoughts from me:

- One thing I like about Read the Docs is that, while it's a private company, they do a good job of aligning themselves with values that many open communities care about. For example, I appreciate the thoughtfulness they've put into [their ethical ads approach](https://docs.readthedocs.io/en/stable/advertising/ethical-advertising.html).
- More generally, I think it's healthy to follow the practice that, where we rely on private or proprietary services, we choose services that:
  - Have a sustainability model that is proven
  - If such a model is not yet proven, they don't rely on VC funding to sustain themselves (or large injections of cash needed to operate)
  - Clearly fit in with a broader communtiy-led ecosystem of tools and standards

That said, there are definitely some benefits to a bigger company like Netlify, its UI/UX is a bit cleaner and it's a bit more flexible.

I also agree for sure that TTW would benefit from a dedicated URL, so that you can decouple "where is the website hosted" from "what is the URL of the website". It reminds me of [this article about the dangers of depending too much on another's platform](https://howtomarketagame.com/2021/11/01/dont-build-your-castle-in-other-peoples-kingdoms/).

**Comment by sgibson91** [comment]
> This can go as an agenda for the next core team meeting next month if we are not in hurry

I don't think this is a time-sensitive decision to make, and it would be better for everyone if time is taken to collect information and deliberate accordingly.

**Comment by bsipocz** [comment]
Why is RTD the other option and not e.g. gh-pages? IMO the latter would be rather convenient, and would kind of work out of the box and very similarly how the other jupyterbook projects work (e.g. while libraries use RTD all the time, I can't really recall any jupyerbook based project that do the same, it's usually render with circleCI, then deploy on gh-pages).

**Comment by bsipocz** [comment]
Netlify currently supports python 3.8 or 2.7, so there is also a slight chance that it may become a problem (though as I see jupyerbook is not rushing to drop old versions, so this won't be an urgent problem)

**Comment by sgibson91** [comment]
I don't think gh-pages matches Turing Way's values in that it is not really an open source build system. Also members of the community have already expressed concerns about the project being hosted on GitHub, rather than GitLab for example, so if we further bake the build/hosting process into GitHub, that further confuses things and makes a migration more difficult if the project does move away from GitHub in the future (I think that's a really, _really_ long-term problem though and not a high priority).

Ideally, I'd love for someone on the project to be tasked with regularly assessing the features of various open source tools and platforms (jupyterbook, mkdocs, hugo, and netlify, RTD, etc) compare them with the features being used by, or desired by the working groups and then make a yearly "stick or switch" recommendation as technologies advance. However, I feel like such a role will have to be paid and so it is a long way off getting there. RTD feels like the option that shares the most of the Turing Way's values, can be enacted now fairly easily, and is much less horrible to work with than Netlify.

**Comment by sgibson91** [comment]
Sorry, I forgot the actually crucial reason in my earlier comment which is deploy previews. Being able to preview changes from PRs (without having to locally build) is crucial to this community and is not a feature that gh-pages offers (as far as I'm aware, I keep looking but haven't found anything yet). RTD supports this and, at the time when I opened this issue, were developing an action that would create a comment with the link to the preview, similar to Netlify's behaviour now. So I that's why I think RTD is the alternative to consider.

I agree that the hosting platform should do as little of the building as possible, and the closer we can get to "feed it html to put on the internet" the better from a portability standpoint.

**Comment by bsipocz** [comment]
My comment above details what a lot of other projects do in CI to provide the preview. It's with circleCI (so addresses the issue of diversification of infrastructure), then the hosting itself is on gh-pages. 
RTD could work fine, or could not. While it's significantly better than netlify, a lot of moving parts behave like a black box that cannot be controlled (or is very clumsy), e.g. the build images, used versions ,etc.
